packages/metadata-service/src/index.ts

```diff
@@ -28,70 +42,199 @@ server.get("/health", async () => {
   return { ok: true };
 });
 
-let program: Program<HeliumEntityManager>;
-server.get<{ Params: { keyToAssetKey: string } }>(
-  "/v1/:keyToAssetKey",
-  async (request, reply) => {
-    program = program || (await init(provider));
-    const { keyToAssetKey } = request.params;
-    const keyToAsset = new PublicKey(keyToAssetKey);
-    const keyToAssetAcc = await program.account.keyToAssetV0.fetch(keyToAsset);
-    const { entityKey, keySerialization } = keyToAssetAcc;
-    const keyStr = decodeEntityKey(entityKey, keySerialization);
-    const digest = animalHash(keyStr);
-    const record = await KeyToAsset.findOne({
+server.get<{ Params: { wallet: string } }>(
+  "/v2/wallet/:wallet",
+  async (request) => {
+    const { wallet } = request.params;
+    let page = 1;
+    const limit = 1000;
+    let allAssets: Asset[] = [];
+
+    while (true) {
+      const assets =
+        (await searchAssets(provider.connection.rpcEndpoint, {
+          ownerAddress: wallet,
+          creatorVerified: true,
+          creatorAddress: entityCreatorKey(daoKey(HNT_MINT)[0])[0].toBase58(),
+          page,
+          limit,
+        })) || [];
+
+      allAssets = allAssets.concat(assets);
+
+      if (assets.length < limit) {
+        break;
+      }
+
+      page++;
+    }
+
+    const assetIds = allAssets.map((a) => a.id);
+    const keyToAssets = await KeyToAsset.findAll({
       where: {
-        address: keyToAsset.toBase58(),
+        asset: assetIds.map((a) => a.toBase58()),
       },
       include: [IotHotspotInfo, MobileHotspotInfo],
     });
 
-    // HACK: If it has a long key, it's an RSA key, and this is a mobile hotspot.
-    // In the future, we need to put different symbols on different types of hotspots
-    const hotspotType = entityKey.length > 100 ? "MOBILE" : "IOT";
-    const image = `${SHDW_DRIVE_URL}/${
-      hotspotType === "MOBILE"
-        ? record?.mobile_hotspot_info?.isActive
-          ? "mobile-hotspot-active.png"
```

Comment:
This logic no longer needed? Looks like its just server hotspot.png and hotspot-active.png